ocs_ci/ocs/node.py

```diff
@@ -110,7 +111,11 @@ def get_nodes(node_type=constants.WORKER_MACHINE, num_of_nodes=None):
             if node_type
             in node.ocp.get_resource(resource_name=node.name, column="ROLES")
         ]
-    if is_hci_provider_cluster() and node_type == constants.WORKER_MACHINE:
+    if (
+        is_hci_provider_cluster()
+        and node_type == constants.WORKER_MACHINE
+        and config.ENV_DATA["worker_replicas"] == 0
```

Comment:
There is a condition in the below for loop to skip master nodes. In compact mode cluster, the worker will also have the master node role. So no nodes will be selected.